src/bindings/build.py

```diff
@@ -34,6 +34,17 @@
     with open(header, "r") as hfile:
         ffi.cdef(hfile.read())
 
+# we need to easily control a few values for Windows builds
+source = []
+if os.getenv("PYNACL_SODIUM_STATIC") is not None:
+    source.append("#define SODIUM_STATIC")
+
+source.append("#include <sodium.h>")
+
+if os.getenv("PYNACL_SODIUM_LIBRARY_NAME") is not None:
+    libraries = [os.getenv("PYNACL_SODIUM_LIBRARY_NAME")]
```

Comment:
Ok, I think just hardcode it to `libsodium` and we can do something
different if someone complains.

On Sat, Jan 9, 2016 at 10:59 AM, Paul Kehrer <notifications@github.com>
wrote:

> In src/bindings/build.py
> <https://github.com/pyca/pynacl/pull/164#discussion_r49265169>:
>
> > @@ -34,6 +34,17 @@
> >      with open(header, "r") as hfile:
> >          ffi.cdef(hfile.read())
> >
> > +# we need to easily control a few values for Windows builds
> > +source = []
> > +if os.getenv("PYNACL_SODIUM_STATIC") is not None:
> > +    source.append("#define SODIUM_STATIC")
> > +
> > +source.append("#include <sodium.h>")
> > +
> > +if os.getenv("PYNACL_SODIUM_LIBRARY_NAME") is not None:
> > +    libraries = [os.getenv("PYNACL_SODIUM_LIBRARY_NAME")]
>
> Since there's no suffix difference for static vs dynamic vs link time
> compile optimization library types Windows people appear to sometimes
> totally rename their libraries. We saw this with the slproweb OpenSSL
> library names before we switched to our own for cryptography. In this case
> there are vcxproj files (at least in the git tree, they don't appear to
> actually ship in the tarball) for the various VS201X releases that probably
> do give the "libsodium" name by default.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/pynacl/pull/164/files#r49265169>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084